### ISSUE TITLE
GeoTag-less backend

### ISSUE THREAD

**Issue by rsimon** [pelagios/recogito2#413]
I think it's worth experimenting with a version of Recogito that removes the "referential integrity mechanism" of the GeoTags. Some future features may not be possible without GeoTags (e.g. in-document place search with autocomplete); and some current features will only be possible with lower performance (data exports, map view). But at the same time:

- GeoTags introduce significant complexity in the backend
- they would need to be refactored/revised, anyway, in order to ease integration of prosopographical lookup functionality

**Comment by rsimon** [comment]
There's also a strong relation to the requirement for "free URIs" (i.e. the ability to put a URI into an annotation that does not refer to a place/person registered in the Recogito index): https://github.com/DigiClass/recogito2/issues/5

**Comment by rsimon** [comment]
Note: this will also make integration of 3rd party NER/Resolution components easier that bring their own URIs. (Currently Recogito would only accept URIs to entities it stores in its index. Once that limitation is relaxed, we can also handle URIs we don't explicitely index ourselves.)

**Comment by rsimon** [comment]
Starting to investigate this more closely. The two methods that are enabled by GeoTags are [listPlacesInDocument](https://github.com/pelagios/recogito2/blob/master/app/services/geotag/GeoTagStore.scala#L39) and [searchPlacesInDocument](https://github.com/pelagios/recogito2/blob/master/app/services/geotag/GeoTagStore.scala#L42).

**Comment by rsimon** [comment]
The architecture change is now implemented and will be deployed as part of the v3.0 release. 

- `listPlacesInDocument` works the way described above, and doesn't seem to have noticable
   performance disadvantages. 
- The size limit (see above) exists, but isn't different from the previous implementation.
- Migration of the live system data was tested locally (cf.
  [documentation on the Wiki](https://github.com/pelagios/recogito2/wiki/Upgrade-to-Recogito-3.0)). For 
  migration of the annotations (the only data type where migration is challenging), I built a tailored "migration utility", `storage.es.migration.AnnotationMigrationUtil`.
- `searchAnnotationsInDocument` is not yet implemented. But looks do-able as a two-phase query. 
  Either as described above, or by first aggregating all referenced entity IDs for the document, and then
  running a second query against the entity index, restricted by these IDs.
